src/java.base/share/classes/sun/security/pkcs/PKCS7.java

```diff
@@ -578,19 +578,18 @@ public SignerInfo verify(SignerInfo info, byte[] bytes)
     public SignerInfo[] verify(byte[] bytes)
     throws NoSuchAlgorithmException, SignatureException {
 
-        Vector<SignerInfo> intResult = new Vector<>();
+        ArrayList<SignerInfo> intResult = new ArrayList<>();
         for (int i = 0; i < signerInfos.length; i++) {
 
             SignerInfo signerInfo = verify(signerInfos[i], bytes);
             if (signerInfo != null) {
-                intResult.addElement(signerInfo);
+                intResult.add(signerInfo);
             }
         }
         if (!intResult.isEmpty()) {
 
             SignerInfo[] result = new SignerInfo[intResult.size()];
-            intResult.copyInto(result);
-            return result;
+            return intResult.toArray(result);
```

Comment:
could be simplified to
`return intResult.toArray(new SignerInfo[0]);`
which would eliminate array zeroing as bonus as I have learned from your other PR ;)

Reply:
I considered this, when I was preparing the patch. But decided to reduce scope of changes.